### PR TITLE
Upgrade mongoose, update graphql query based on registration schema

### PR DESCRIPTION
**Pull request by kexin-zhang**
Resolves errors that we talked about earlier today: 
* Upgraded mongoose to v5 
* Switched out the `attending` field in `UserFilter` for `confirmed` to match the current registration graphQL schema

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ app.use(morgan("hackgt"));
 
 (mongoose as any).Promise = global.Promise;
 mongoose.connect(MONGO_URL, {
-	useMongoClient: true
+	useNewUrlParser: true
```
Comment (ehsanmasdar) [inline server/app.ts]: Why is this necessary?

Comment (kexin-zhang) [reply server/app.ts]: I'm not sure if it's necessary, but I did it to get rid of this warning:
```
(node:13240) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.
```
